[GPUP] The system language in the Web Process needs to match the system language in the GPU Process

litherum opened the pull request:
#### 50f8feea3d4e1a756d156a56519d1c33a8fdad92
<pre>
[GPUP] The system language in the Web Process needs to match the system language in the GPU Process
<a href="https://bugs.webkit.org/show_bug.cgi?id=242581">https://bugs.webkit.org/show_bug.cgi?id=242581</a>
&lt;rdar://problem/94135218&gt;

Reviewed by Chris Dumez.

If the Web Process and the GPU process disagree about the system language, we can&apos;t IPC
system fonts correctly.

This patch simply moves the code that sets the web process&apos;s system language from WebProcessProxy
to AuxiliaryProcessProxy, so all processes get it, so all WebKit processes agree on the system language.

Test: fast/text/international/system-language/han-text-style.html

* Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp:
(WebKit::AuxiliaryProcessProxy::populateOverrideLanguagesLaunchOptions):
(WebKit::AuxiliaryProcessProxy::getLaunchOptions):
(WebKit::AuxiliaryProcessProxy::platformOverrideLanguages const):
* Source/WebKit/UIProcess/AuxiliaryProcessProxy.h:
* Source/WebKit/UIProcess/Cocoa/AuxiliaryProcessProxyCocoa.mm:
(WebKit::AuxiliaryProcessProxy::platformOverrideLanguages const):
* Source/WebKit/UIProcess/Cocoa/WebProcessProxyCocoa.mm:
(WebKit::WebProcessProxy::platformOverrideLanguages const): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::getLaunchOptions):
(WebKit::WebProcessProxy::fullKeyboardAccessEnabled):
(WebKit::WebProcessProxy::platformOverrideLanguages const): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.h:

Canonical link: <a href="https://commits.webkit.org/252347@main">https://commits.webkit.org/252347@main</a>
</pre>

cdumez commented on Source/WebKit/UIProcess/AuxiliaryProcessProxy.h:
```diff
@@ -185,6 +185,9 @@ class AuxiliaryProcessProxy : public ThreadSafeRefCounted<AuxiliaryProcessProxy,
     virtual void connectionWillOpen(IPC::Connection&);
     virtual void processWillShutDown(IPC::Connection&) = 0;
 
+    void populateOverrideLanguagesLaunchOptions(ProcessLauncher::LaunchOptions&);
```
Cannot this function be marked const?

litherum replied:
Even better, I think it can be moved outside of the class and just be a static standalone function! 😁

cdumez replied:
Yes, static would be better indeed (inside or outside the class).

litherum replied:
Oh, wait, it calls `platformOverrideLanguages()` which is a `virtual` function, so it can't be `static` 😔

`const` it is!

cdumez reviewed (approved):
LGTM

litherum committed:
[GPUP] The system language in the Web Process needs to match the system language in the GPU Process
https://bugs.webkit.org/show_bug.cgi?id=242581
<rdar://problem/94135218>

Reviewed by Chris Dumez.

If the Web Process and the GPU process disagree about the system language, we can't IPC
system fonts correctly.

This patch simply moves the code that sets the web process's system language from WebProcessProxy
to AuxiliaryProcessProxy, so all processes get it, so all WebKit processes agree on the system language.

Test: fast/text/international/system-language/han-text-style.html

* Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp:
(WebKit::AuxiliaryProcessProxy::populateOverrideLanguagesLaunchOptions):
(WebKit::AuxiliaryProcessProxy::getLaunchOptions):
(WebKit::AuxiliaryProcessProxy::platformOverrideLanguages const):
* Source/WebKit/UIProcess/AuxiliaryProcessProxy.h:
* Source/WebKit/UIProcess/Cocoa/AuxiliaryProcessProxyCocoa.mm:
(WebKit::AuxiliaryProcessProxy::platformOverrideLanguages const):
* Source/WebKit/UIProcess/Cocoa/WebProcessProxyCocoa.mm:
(WebKit::WebProcessProxy::platformOverrideLanguages const): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::getLaunchOptions):
(WebKit::WebProcessProxy::fullKeyboardAccessEnabled):
(WebKit::WebProcessProxy::platformOverrideLanguages const): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.h:

Canonical link: https://commits.webkit.org/252347@main
